services/src/handlers/datasets.rs

```diff
@@ -555,15 +632,41 @@ fn detect_time_type(columns: &Columns) -> OgrSourceDatasetTimeType {
     }
 }
 
-fn detect_vector_type(layer: &Layer) -> Result<VectorDataType> {
-    let ogr_type = layer
-        .defn()
-        .geom_fields()
-        .next()
-        .context(error::EmptyDatasetCannotBeImported)?
-        .field_type();
+fn detect_vector_geometry(dataset: &Dataset) -> DetectedGeometry {
+    if let Ok(layer) = dataset.layer(0) {
```

Comment:
for each?

Reply:
87b15682f1e70e0687e5cb55d2d11478bc28fcf7